examples/serenity/postgres/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+use serenity::async_trait;
+use serenity::model::prelude::*;
+use serenity::prelude::*;
+use shuttle_service::error::CustomError;
+use sqlx::{Executor, FromRow, PgPool};
+use std::env;
+
+struct Bot {
+    database: PgPool,
+}
+
+#[derive(FromRow)]
+struct Todo {
+    pub id: i32,
+    pub note: String,
+}
+
+#[async_trait]
+impl EventHandler for Bot {
+    async fn message(&self, ctx: Context, msg: Message) {
+        // The user_id of the user sending a command
+        let user_id = msg.author.id.0 as i64;
+
+        // Add a new todo using `~todo add <note>` and persist it in postgres
+        if let Some(note) = msg.content.strip_prefix("~todo add") {
```

Comment:
I wonder if it is worth modelling this example after the [slash commands](https://github.com/serenity-rs/serenity/blob/current/examples/e14_slash_commands/src/main.rs) to allow invoking these with `/todo add` instead of `~todo add`?